security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -154,55 +163,63 @@ func (s *sdsservice) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecre
 				log.Errorf("Failed to parse discovery request: %v", err)
 				return err
 			}
+
 			con.proxyID = discReq.Node.Id
 			con.ResourceName = resourceName
 
-			log.Debugf("Received SDS request from %q, resourceName %q, versionInfo %q\n", discReq.Node.Id, resourceName, discReq.VersionInfo)
+			key := cache.ConnKey{
+				ResourceName: resourceName,
+			}
+			con.mutex.Lock()
+			if con.conID == "" {
```

Comment:
Should we log something here, maybe in debug mode for whether it is the first or now?

Reply:
add a flag and print different debug logs below